AN-7761 Fix broken cldr require for some languages

thallada opened the pull request:
This issue was preventing many users from loading parts of Insights. ~~I added a post-install task to rename cldr-data language directories that had mixed case to fully lowercase names. We already lowercase the language code that we get from `window.language`.~~ Update: Turns out that lowercasing all of the cldr-data locale directories did break things... Now we are converting the lowercase locales that the browser reports to the casing that cldr-data expects by doing a mapping in the javascript code instead. Much cleaner.

I tested this with my chrome browser set to `en-gb` and verified that the pages now load.

~~As far as I can tell, nothing else in the application is affected by lowercasing all of the cldr directory names.~~

I also did some improvements to globalization like use the cldr `availableLocales` instead of a static list, added a test for lowercasing the window.language, defaulted to "en" when undefined, and silenced all "Copying" log messages from `bower_components/cldr-data/main/*` during collectstatic (was causing Travis to error out because of the sheer number of lines of logs it produces).

dsjen commented on bower-post-install.sh:
```diff
@@ -19,6 +19,18 @@ CSS2SCSS ${BOWER_COMPONENTS_PATH}/nvd3/build/nv.d3
 
+# Rename lang folders to be fully lowercase (see AN-7761)
+mv_counter=0
+for i in analytics_dashboard/static/bower_components/cldr-data/main/*; do
```
We could use ${BOWER_COMPONENTS_PATH}/cldr-data/main/*.  Maybe a bit DRYer?

thallada replied:
ah yes, that is cleaner!

thallada commented:
Ok! Addressed your comments, @dsjen. If you could delete your `/static/bower_components/cldr-data` directory and try running `make develop` again to test that would be helpful! Thanks!

dsjen commented:
There must be something different about the mac!  I'm not seeing the directories renamed to lowercase oddly.  I'll try to see what's amiss in a bit.

thallada commented:
So, we found out that I didn't actually test this fully. Globalize does not like the lang folders being lowercase. I realized there was a better solution to this: with `availableLocales` we can create a mapping of those values lowercased to the original values. Then, in `globalization.js` we can map whatever we get from the browser to the correctly-cased value that CLDR expects. @dsjen, could you delete the cldr-data folder, bower install again, and test this? (for the last time, hopefully!)

dsjen commented:
Fun times debugging!  🐛   I tested this out on a few languages and confirms it works.  👍 

Also, just in case you weren't thinking of it already, please squash the commits.

thallada commented:
Good to hear! Squashed! :hammer:

Once the tests merge I'll merge.
